### PR TITLE
Cdhit dup lzw step

### PR DESCRIPTION
**Pull request by yunfangjuan**
- CDHitDup
- LZW 
etc

### CODE REVIEW

File: idseq_dag/steps/run_lzw.py [hunk 1]
```diff
@@ -1,8 +1,126 @@
...
 
 class PipelineStepRunLZW(PipelineStep):
+
     def run(self):
-        for f in self.output_files_local():
-            command.execute("date > %s" % f)
+        input_fas = self.input_files_local[0]
+        output_fas = self.output_files_local()
+        cutoff_fractions = self.additional_attributes["thresholds"]
+        PipelineStepRunLZW.generate_lzw_filtered(input_fas, output_fas, cutoff_fractions)
```
Comment (jshoe) [inline idseq_dag/steps/run_lzw.py]: would you rather 'PipelineStepRunLZW' be 'self'?

File: idseq_dag/steps/run_lzw.py [hunk 2]
```diff
@@ -1,8 +1,126 @@
...
+        for os in outstream_list:
+            for o in os:
+                o.close()
+        for r in read_streams:
+            r.close()
+        # get the right output file and metrics
+        cutoff_frac = cutoff_fractions[-1]
+        kept_count = 0
+        filtered = total_reads
+        for i in range(len(readcount_list)):
```
Comment (tfrcarvalho) [inline idseq_dag/steps/run_lzw.py]: You can be more pythonic: :)
`for i, readcount in enumerate(readcount_list):`

File: idseq_dag/steps/run_lzw.py [hunk 3]
```diff
@@ -1,8 +1,126 @@
...
+            readcount = readcount_list[i]
+            if readcount > 0:
+                # found the right bin
+                cutoff_frac = cutoff_fractions[i]
+                kept_count = readcount
+                filtered = total_reads - kept_count
+                outfiles = outfiles_list[i]
+                # move the output files over
+                for j in range(len(outfiles)):
```
Comment (tfrcarvalho) [inline idseq_dag/steps/run_lzw.py]: Also more pythonic:
`for outfile, output_file in zip(outfiles, output_files):`

File: idseq_dag/steps/run_lzw.py [hunk 4]
```diff
@@ -1,8 +1,126 @@
...
+    @staticmethod
+    def lzw_fraction(sequence):
+        sequence = str(sequence)
+        if sequence == "":
+            return 0.0
+        sequence = sequence.upper()
+        dict_size = 0
+        dictionary = {}
+        # Initialize dictionary with single char
+        for c in sequence:
```
Comment (tfrcarvalho) [inline idseq_dag/steps/run_lzw.py]: Also. more pythonic:
`dictionary = {c: i for i, c in enumerate(sequence)}`

File: idseq_dag/steps/run_lzw.py (hunk 1 above)
Comment (yunfangjuan) [reply idseq_dag/steps/run_lzw.py]: yeah. That's what the linter told me ;-)

File: idseq_dag/steps/run_lzw.py (hunk 2 above)
Comment (yunfangjuan) [reply idseq_dag/steps/run_lzw.py]: thanks! i am a python newbie.

**Pull request merged**